### PR TITLE
Use warnings.warn instead of logger.warning

### LINKED ISSUE #319: WARNING this function is deprecated, use smart_open.open instead
Every operation I do with gensim I am seeing "2019-05-23 09:42:54,149 WARNING this function is deprecated, use smart_open.open instead"

for example:
~~~
model = LdaModel.load(os.path.join(TEMP_FOLDER, 'lda.model'))
model.print_topics(10)
~~~

I guess it started after upgrading gensim. 

smart-open        1.8.3
gensim            3.7.3

### PR DESCRIPTION
**Pull request by mpenkov**
Fix #319

### CODE REVIEW

File: smart_open/smart_open_lib.py [hunk 1]
```diff
@@ -382,7 +382,7 @@ def open(
 
 def smart_open(uri, mode="rb", **kw):
     """Deprecated, use smart_open.open instead."""
-    logger.warning('this function is deprecated, use smart_open.open instead')
+    warnings.warn('this function is deprecated, use smart_open.open instead')
```
Comment (piskvorky) [inline smart_open/smart_open_lib.py]: Can we link to the `smart_open` => `open` migration instructions here? This "use smart_open.open instead" is a massive ask, unless it's a 1:1 swap without any potential issues (which I don't think it is).

Let's inform the user more clearly what are the expected tradeoffs / benefits of such change, and what their migration and potential incompatibilities look like.

File: README.rst [hunk 2]
```diff
@@ -255,9 +255,113 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
...
   ...     print(key, round(len(content) / 2**20))
-  annual/monthly_rain/2010.monthly_rain.nc 14
-  annual/monthly_rain/2011.monthly_rain.nc 14
-  annual/monthly_rain/2012.monthly_rain.nc 14
+  annual/monthly_rain/2010.monthly_rain.nc 13
+  annual/monthly_rain/2011.monthly_rain.nc 13
+  annual/monthly_rain/2012.monthly_rain.nc 13
+
+
+Migrating to the New open Function
```
Comment (piskvorky) [inline README.rst]:
```suggestion
Migrating to the new `open` function
```

File: README.rst [hunk 3]
```diff
@@ -255,9 +255,113 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
...
+----------------------------------
+
+Since 1.8.1, there is a `smart_open.open` function that replaces `smart_open.smart_open`.
+The new function offers several advantages over the old one:
+
+- 100% compatible with the built-in open function (aka io.open): it accepts all
```
Comment (piskvorky) [inline README.rst]:
```suggestion
- 100% compatible with the built-in `open` function (aka `io.open`): it accepts all
```

File: README.rst [hunk 4]
```diff
@@ -255,9 +255,113 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
...
+  the parameters that the built-in open accepts.
+- Default open mode is now "r", the same as for the built-in open
```
Comment (piskvorky) [inline README.rst]:
```suggestion
- The default `smart_open.open` mode is now "r", the same as for the built-in `open` (`smart_open.smart_open` default used to be "rb").
```

File: README.rst [hunk 5]
```diff
@@ -255,9 +255,113 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
...
+- Fully documented keyword parameters (try `help("smart_open.open")`)
+
+These instructions will help you migrate to the new function painlessly.
```
Comment (piskvorky) [inline README.rst]:
```suggestion
The instructions below will help you migrate to the new function painlessly.
```

File: smart_open/smart_open_lib.py [hunk 6]
```diff
@@ -380,9 +380,23 @@ def open(
...
-    logger.warning('this function is deprecated, use smart_open.open instead')
+    """Deprecated, use smart_open.open instead.
+
+    See the migration notes for details: %s
+
+    """ % _MIGRATION_NOTES_URL
+
+    warnings.warn(
+        'This function is deprecated, use smart_open.open instead. '
+        'See the migration notes for details: %s' % _MIGRATION_NOTES_URL
```
Comment (piskvorky) [inline smart_open/smart_open_lib.py]:
```suggestion
        'See the migration instructions: %s' % _MIGRATION_NOTES_URL
```

File: README.rst [hunk 7]
```diff
@@ -255,9 +255,114 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
...
   ...     print(key, round(len(content) / 2**20))
-  annual/monthly_rain/2010.monthly_rain.nc 14
-  annual/monthly_rain/2011.monthly_rain.nc 14
-  annual/monthly_rain/2012.monthly_rain.nc 14
+  annual/monthly_rain/2010.monthly_rain.nc 13
+  annual/monthly_rain/2011.monthly_rain.nc 13
+  annual/monthly_rain/2012.monthly_rain.nc 13
+
+
+Migrating to the New `open` Function
```
Comment (piskvorky) [inline README.rst]: @mpenkov what happened here? I remember commenting on this line in a review, and you fixing it. But now it's back in the README?

File: README.rst [hunk 8]
```diff
@@ -255,9 +255,114 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
...
+Renamed parameters:
+
+- `s3_upload` ➡  `multipart_upload_kwargs`
+- `s3_session` ➡ `session`
+
+Removed parameters:
+
+- `profile_name`
+
+**The `profile_name` parameter has been removed.**
```
Comment (piskvorky) [inline README.rst]: Something is wrong with the backticks. They have no effect in this section (look like normal text when rendered), and this line actually renders them as literal backticks:

Screen Shot 2019-05-31 at 09 58 03

File: README.rst [hunk 9]
```diff
@@ -255,9 +255,114 @@ Since going over all (or select) keys in an S3 bucket is a very common operation
...
+  >>> url = 's3://smart-open-py37-benchmark-results/test.txt'
+  >>> session = boto3.Session(profile_name='smart_open')
+  >>> params = {'session': session}
+  >>> open(url, 'r', transport_params=params).read(32)
+  'first line\nsecond line\nthird lin'
+
+See `help("smart_open.open")` for the full list of acceptable parameter names,
+or view the help online `here <https://github.com/RaRe-Technologies/smart_open/blob/master/help.txt>`__.
+
+If you pass an invalid parameter names, the `open` function will warn you about it.
```
Comment (piskvorky) [inline README.rst]:
```
If you pass an invalid parameter name, the `open` function will warn you about it.
```

File: README.rst (hunk 8 above)
Comment (mpenkov) [reply README.rst]: I think that's just the way github renders RST. I don't know what we can do about that.

Comment (piskvorky) [reply README.rst]: Aaah, this is RST, not MD! That explains it.

Literal blocks in RST use double-backticks IIRC, not single-backtick.

Do we need to use RST though? MD easier? Or is it something to do with PyPI relying on RST…

Comment (mpenkov) [reply README.rst]: Yes, it's because of PyPI

**Pull request merged**